.github/workflows/buildapp.yml

```diff
@@ -0,0 +1,189 @@
+# Original idea by @ISnackable. Thanks to him for handling the hardest parts!
+# https://github.com/ISnackable/YTCubePlus/blob/main/.github/workflows/Build.yml
+
+name: Build and Release BHInsta
+
+on:
+  workflow_dispatch:
+    inputs:
+      sdk_version:
+        description: "iOS SDK version to be used during build"
+        default: "14.5"
+        required: true
+        type: string
+      decrypted_instagram_url:
+        description: "The direct URL to the decrypted Instagram ipa"
+        default: "https://download.starfiles.co/ba8d578a5c2f"
+        required: true
+        type: string
+      bundle_id:
+        description: "Modify the bundle ID. Not recommended"
+        default: "com.burbn.instagram"
+        required: true
+        type: string
+      app_name:
+        description: "Modify the name of the app on the Home Screen. Not recommended"
+        default: "Instagram"
+        required: true
+        type: string
+      create_release:
+        description: "Create a draft release"
+        default: true
+        required: false
+        type: boolean
+      upload_artifact:
+        description: "Upload Artifact"
+        default: true
+        required: false
+        type: boolean
+  
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  build:
+    name: Build BHInsta
+    runs-on: macos-latest
+    permissions:
+      contents: write
+
+    steps:
+      - name: Checkout Main
+        uses: actions/checkout@v4.1.1
+        with:
+          path: main
+          submodules: recursive
+
+      - name: Install Dependencies
+        run: brew install ldid dpkg make
+
+      - name: Set PATH environment variable
+        run: echo "$(brew --prefix make)/libexec/gnubin" >> $GITHUB_PATH 
+
+      - name: Setup Theos
+        uses: actions/checkout@v4.1.1
+        with:
+          repository: theos/theos
+          ref: master
+          path: theos
+          submodules: recursive
+      
+      - name: SDK Caching
+        id: SDK
+        uses: actions/cache@v4.0.2
+        env:
+          cache-name: iOS-${{ inputs.sdk_version }}-SDK
+        with:
+          path: theos/sdks/
+          key: ${{ env.cache-name }}
+          restore-keys: ${{ env.cache-name }}
+
+      - name: Download iOS SDK
+        if: steps.SDK.outputs.cache-hit != 'true'
+        run: |
+          git clone --quiet -n --depth=1 --filter=tree:0 https://github.com/arichornlover/sdks/
```

Comment:
If possible it'd be great to download the SDKs from https://github.com/xybp888/iOS-SDKs